sormas-backend/src/main/java/de/symeda/sormas/backend/environment/EnvironmentFacadeEjb.java

```diff
@@ -301,6 +314,36 @@ public static EnvironmentReferenceDto toReferenceDto(Environment entity) {
 		return new EnvironmentReferenceDto(entity.getUuid(), entity.getEnvironmentName());
 	}
 
+	@Override
+	@RightsAllowed(UserRight._ENVIRONMENT_ARCHIVE)
+	public void archive(String entityUuid, Date endOfProcessingDate) {
+		super.archive(entityUuid, endOfProcessingDate);
+	}
+
+	@Override
+	@RightsAllowed(UserRight._ENVIRONMENT_ARCHIVE)
+	public void dearchive(List<String> entityUuids, String dearchiveReason) {
+		super.dearchive(entityUuids, dearchiveReason);
+	}
+
+	@Override
+	@RightsAllowed(UserRight._ENVIRONMENT_DELETE)
+	public void delete(String uuid, DeletionDetails deletionDetails) throws ExternalSurveillanceToolRuntimeException {
+		Environment environment = service.getByUuid(uuid);
+
+		if (!service.inJurisdictionOrOwned(environment)) {
+			throw new AccessDeniedException(I18nProperties.getString(Strings.messageEventParticipantOutsideJurisdictionDeletionDenied));
+		}
+
+		service.delete(environment, deletionDetails);
+	}
+
+	@Override
+	@RightsAllowed(UserRight._ENVIRONMENT_DELETE)
+	public void restore(String uuid) {
+		super.restore(uuid);
+	}
+
 	@Override
 	protected void pseudonymizeDto(Environment source, EnvironmentDto dto, Pseudonymizer pseudonymizer, boolean inJurisdiction) {
```

Comment:
`pseudonymizeDto` and `restorePseudonymizedDto` should be implemented

Reply:
implemented